layer/layer.go

```diff
@@ -173,7 +173,7 @@ type Store interface {
 	Map() map[ChainID]Layer
 	Release(Layer) ([]Metadata, error)
 
-	CreateRWLayer(id string, parent ChainID, mountLabel string, initFunc MountInit, storageOpt map[string]string) (RWLayer, error)
+	CreateRWLayer(id string, parent ChainID, mountLabel string, initFunc MountInit, storageOpt map[string]string, sharedRootFs bool) (RWLayer, error)
```

Comment:
@dmcgowan Moving some of the options in a options struct sounds simpler to do. So I will move `mountLabel`, `initFunc`, `storageOpts` and `sharedRootFs` in a options struct.